processor/src/main/java/org/mapstruct/ap/internal/model/BeanMappingMethod.java

```diff
@@ -298,38 +297,15 @@ else if ( reportErrorOnTargetObject( mapping ) ) {
 
         private void handleDefinedNestedTargetMapping(Set<String> handledTargets) {
 
-            Map<PropertyEntry, MappingOptions> optionsByNestedTarget =
-                method.getMappingOptions().groupByPoppedTargetReferences();
-            for ( Entry<PropertyEntry, MappingOptions> entryByTP : optionsByNestedTarget.entrySet() ) {
+            NestedTargetPropertyMappingHolder holder = new NestedTargetPropertyMappingHolder.Builder()
```

Comment:
I am not sure. I think that we need the "old" routine. The only thing is that we need to consider non-nested for creating the `PropertyMapping` in case we have just renaming like:

```java
@Mapping(target = "quality.document", source = "quality.report"),
@Mapping(target = "quality.document.organisation", source = "quality.report.org")
```